src/main/kotlin/com/hubermjonathan/discord/common/models/Feature.kt

```diff
@@ -4,29 +4,32 @@ import com.hubermjonathan.discord.common.DiscordLogger
 import net.dv8tion.jda.api.JDA
 import java.util.Timer
 
-abstract class Feature(jda: JDA, name: String, icon: String) {
+abstract class Feature(private val jda: JDA, private val name: String, icon: String) {
     abstract val buttons: List<Button>
     abstract val commands: List<Command>
     abstract val managers: List<Manager>
     abstract val tasks: List<Task>
     protected val context = FeatureContext(jda, logger = DiscordLogger(jda, name, icon), name, icon)
 
-    fun load(jda: JDA) {
+    fun load() {
+        val timer = Timer()
         val listenerAdapters = buttons + commands + managers
 
+        commands.forEach { command ->
+            jda.guilds.forEach { guild ->
+                guild.upsertCommand(command.commandData)
+            }
```

Comment:
This isn't doing _exactly_ the same thing as the prior code in `MitchBot.kt`, but I'm fairly confident that it is functionally equivalent.

Whether or not each feature should add its own commands is up for interpretation; however, I argue that they should, since you already set up the event listeners in here.